ts/views/conversation_view.ts

```diff
@@ -1535,6 +1537,15 @@ Whisper.ConversationView = Whisper.View.extend({
     }
   },
 
+  onCopy(e: any) {
+    const selection = window.getSelection();
+    if (selection) {
+      clipboard.writeText(selection.toString());
```

Comment:
1. Unfortunately, this doesn't work for messages that contain emoji. For example, try copying the message "Hello 🌍 world". Is there a way to make that work? (It seems to be an issue with `selection.toString()`, not `clipboard.writeText`.)
2. I'd prefer to use [`navigator.clipboard.writeText`](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText) if possible. Could you update to use the native browser API?

Reply:
Thanks for the feedback. I'll see if I can find a solution for the emojis.

Reply:
Retrieving the emojis are kind of annoying as we have to pick out the text versions from one of the element attributes.
Even though it feels somewhat dirty, cloning up a document fragment and iteratively replacing every emoji image-tag with their alt-attribute values seem to work pretty good from what I can tell. We're not doing anything with the fragment afterwards so changing its contents seem safe enough for me. This is the easiest way to handle this that I've found so far.

But when working with the document fragment instead, we're picking up lots of elements in our selection that aren't really useful for users to copy (e.g. hover texts, menu descriptions, message reactions etc). Therefore I figured it'd be better to only handle specific elements based on their classes such as the message contents, authors and sent-timestamps.

Let me know what you think. I'm not too sure myself if something like this is worth the code complexity, but at the same time I personally don't see how this can be done in any simpler manner while also avoiding cases where the user gets strange or bad clipboard data. Also, there are certainly other improvements that can be done here. One-on-one conversations will typically not show any author names for incoming messages as they're not present in the DOM.

I'll see if I can improve it further and also get some test coverage on this as well if you think something like this could be useful.

Reply:
Copying and pasting the following contents:
![bilde](https://user-images.githubusercontent.com/79846697/126566987-254d0d0d-fc0a-4a6e-826a-d79e9ab436c5.png)

Would give this ("Du" meaning "You" in Norwegian):

Du, 11m:
Some 😌🙏 emojis 😳
Du, 8m:
👀👉 In a note to myself as I can't get more 😤 standalone clients up and 🏃‍♂‍
Du, 7m:
A reply to myself as well